### DISCUSSION TITLE
Possible to support pydantic

### DISCUSSION THREAD

**Discussion by michaeloliverx** [microsoft/pylance-release#1013] [Ideas] [answered]
Would it be possible to provide support for [`pydantic`](https://github.com/samuelcolvin/pydantic) `BaseModel`. pydantic autogenerates the `__init__` for classes similar to dataclass/attrs. See [this](https://github.com/samuelcolvin/pydantic/discussions/2464) discussion on the pydantic repository for a better description.

**Comment by erictraut** [comment, marked as answer]
Pyright, the type checker upon which pylance is based, uses standard static type information provided by libraries. It does not (nor do we have any plans to) provide any special casing for third-party libraries like pydantic.

We have given some thought to the possibility of creating a way to extend static typing so library authors can specify more dynamic behaviors. This is a significant undertaking, and if it does happen, we need to go through an open specification and standardization process so all type checkers and all libraries benefit. Just to set expectations, this is not something that will happen any time soon.

In the meantime, libraries that use dynamic behaviors are not going to work well with static type checking.
